bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateProgressInfo.java

```diff
@@ -72,37 +72,81 @@ protected FirmwareUpdateProgressInfo() {
+            ThingTypeUID thingTypeUID, String firmwareVersion, ProgressStep progressStep,
+            Collection<ProgressStep> sequence, boolean pending, int progress) {
+        return new FirmwareUpdateProgressInfo(thingUID, thingTypeUID, firmwareVersion, progressStep, sequence, pending,
+                progress);
+    }
+
+    private FirmwareUpdateProgressInfo(ThingUID thingUID, ThingTypeUID thingTypeUID, String firmwareVersion,
+            ProgressStep progressStep, Collection<ProgressStep> sequence, boolean pending) {
+        Objects.requireNonNull(thingUID, "ThingUID must not be null.");
+        Objects.requireNonNull(thingTypeUID, "ThingTypeUID must not be null.");
+        Objects.requireNonNull(firmwareVersion, "Firmware version must not be null.");
+
         if (sequence == null || sequence.isEmpty()) {
             throw new IllegalArgumentException("Sequence must not be null or empty.");
         }
         Objects.requireNonNull(progressStep, "Progress step must not be null.");
 
-        this.firmwareUID = firmwareUID;
+        this.thingUID = thingUID;
+        this.thingTypeUID = thingTypeUID;
+        this.firmwareVersion = firmwareVersion;
         this.progressStep = progressStep;
         this.sequence = sequence;
         this.pending = pending;
         this.progress = null;
     }
 
     /**
-     * Returns the UID of the firmware that is updated.
+     * Creates a new {@link FirmwareUpdateProgressInfo}.
+     *
+     * @param thingTypeUID the thing type UID of the firmware that is updated (must not be null)
+     * @param firmwareVersion the version of the firmware that is updated (must not be null)
+     * @param progressStep the current progress step (must not be null)
+     * @param sequence the collection of progress steps describing the sequence of the firmware update process
+     *            (must not be null)
+     * @param pending the flag indicating if the update is pending
+     * @return FirmwareUpdateProgressInfo object (not null)
+     * @throws NullPointerException if thingUID, thingTypeUID, firmwareVersion or current progress step is null
+     * @throws IllegalArgumentException if sequence is null or empty
+     */
+    @NonNull
+    public static FirmwareUpdateProgressInfo createFirmwareUpdateProgressInfo(ThingUID thingUID,
+            ThingTypeUID thingTypeUID, String firmwareVersion, ProgressStep progressStep,
+            Collection<ProgressStep> sequence, boolean pending) {
+        return new FirmwareUpdateProgressInfo(thingUID, thingTypeUID, firmwareVersion, progressStep, sequence, pending);
+    }
+
+    /**
+     * Returns the Thing type UID of the firmware that is updated.
      *
-     * @return the UID of the firmware that is updated (not null)
+     * @return the Thing type UID ofthe firmware that is updated (not null)
      */
-    public FirmwareUID getFirmwareUID() {
-        return firmwareUID;
+    public ThingTypeUID getThingTypeUID() {
```

Comment:
do we need to expose this information?

Reply:
The same as above, the thing type UID can be found by the thing UID so I will remove it from here as well